dagshub/common/helpers.py

```diff
@@ -31,3 +40,99 @@ def http_request(method, url, **kwargs):
         if arg not in kwargs:
             kwargs[arg] = mixin_args[arg]
     return httpx.request(method, url, **kwargs)
+
+
+def get_project_root(root):
+    while not (root / '.git').is_dir():
+        if ismount(root):
+            raise ValueError('No git project found! (stopped at mountpoint {root}). \
+                             Please run this command in a git repository.')
+        root = root / '..'
+    return Path(root)
+
+
+def _init(repo_name=None, repo_owner=None, url=None, root=None,
+          host=config.DEFAULT_HOST, mlflow=True, dvc=False):
+    # Setup required variables
+    root = root or get_project_root(Path(os.path.abspath('.')))
```

Comment:
Wrap in a try-catch and show the error message instead of a scary stack trace.

Also if root is specified check that it is actually a git repo